### PR TITLE
feat: support TRUNCATE

### PR DESCRIPTION
**Pull request by olavloite**
Add support for the `TRUNCATE` statement. This is translated into one or more `DELETE` statements. The statement can be combined with the `spanner.autocommit_dml_mode` setting to make the statement non-atomic and usable for tables that contain a large number of rows.

go/pgadapter-truncate

### CODE REVIEW

File: src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+  static ParsedTruncateStatement parse(String sql) {
+    Preconditions.checkNotNull(sql);
+
+    SimpleParser parser = new SimpleParser(sql);
+    ParsedTruncateStatement.Builder builder = new Builder();
+    if (!parser.eatKeyword("truncate")) {
+      throw PGExceptionFactory.newPGException("not a valid TRUNCATE statement: " + sql);
+    }
+    parser.eatKeyword("table");
```
Comment (pratickchokhani) [inline src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java:130]: Should we make this a static string to reduce memory?

File: src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    builder.only = parser.eatKeyword("only");
```
Comment (pratickchokhani) [inline src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java:131]: Same.

File: src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    builder.star = parser.eatToken("*");
+    while (parser.eatToken(",")) {
+      TableOrIndexName name = parser.readTableOrIndexName();
+      if (name == null) {
+        throw PGExceptionFactory.newPGException("invalid or missing table name");
+      }
+      builder.tables.add(name);
+    }
+    if (!parser.eatKeyword("continue", "identity")) {
+      builder.restartIdentity = parser.eatKeyword("restart", "identity");
```
Comment (pratickchokhani) [inline src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java:146]: Same.

**Review by pratickchokhani — APPROVED** [review]

File: src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java (hunk 1 above)
Comment (olavloite) [reply src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java:130]: That is actually not necessary in Java. String literals are automatically 'interned' (added to the internal String pool of the JVM), which means that repeating the same string literal throughout an application will not use more memory than creating a single String constant and reuse that. See https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#intern--

File: src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java (hunk 2 above)
Comment (olavloite) [reply src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java:131]: See above

File: src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java (hunk 3 above)
Comment (olavloite) [reply src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java:146]: See above

**Pull request merged**